lightning/src/chain/channelmonitor.rs

```diff
@@ -1680,8 +1682,7 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitor<Signer> {
 	/// missed/unhandled descriptors. For the purpose of gathering historical records, if the
 	/// channel close has fully resolved (i.e., [`ChannelMonitor::get_claimable_balances`] returns
 	/// an empty set), you can retrieve all spendable outputs by providing all descendant spending
-	/// transactions starting from the channel's funding or closing transaction that have at least
-	/// [`ANTI_REORG_DELAY`] confirmations.
+	/// transactions starting from the channel's funding transaction and going down three levels.
```

Comment:
Are these three transactions the three levels?
1. Funding tx
2. Closing tx
3. Tx spending the closing tx outputs

Reply:
Makes sense that the funding transaction isn't needed. Why is the third level needed? Isn't the closing transaction and all spends of the closing transaction enough?

Reply:
Oh that's right, there could be the revocation tx as well! 

So this scenario would be:
1. Commitment tx (closing tx)
2. Revocation tx (spends the commitment tx)
3. Tx spending the revocation tx

We need the third layer because we want to make sure that the output is spent right?

Thanks for the speedy reply btw 🙏

Reply:
> Because HTLCs are resolved in two stages when you broadcast your own commitment transaction.

I didn't fully grok the two stage HTLC protocol until now. I still only see two layers that we need to check? I.e
1. Commitment Tx
2. HTLC-timeout Tx (which contains the LDK spendable output)

<img width="723" alt="Screenshot 2023-10-30 at 2 15 57 PM" src="https://github.com/lightningdevkit/rust-lightning/assets/5805570/c696e484-ddfb-4832-a6dd-8cf47965c78a">

Reply:
Gotcha, that makes sense